### ISSUE TITLE
Coloring by an arbitrary cell metadata column (enhancement)

### ISSUE THREAD

**Issue by VPetukhov** [euxhenh/cellar#9]
To my understanding, currently there is no way to color the dataset by anything but cluster id. Having a color column selector would add a lot to usability. For example, adding a drop-down to the left of "[Change Color Palette](https://github.com/euxhenh/cellar/blob/a1b9e551ed1ba29a7808df9dc9cc4359aae27008/layout/plots.py#L170)". Seems like a very easy fix. Do you plan adding such functionality? If it would help, I can submit a PR for that.

**Comment by euxhenh** [comment]
I agree that this would be useful. Were you thinking of allowing all keys in `adata.obs` to be selected? How would we handle columns with too many categorical values? From my experience, this can slow down plot generation a lot.

**Comment by VPetukhov** [comment]
Thank you for the quick reply! Yes, I meant all variables from `adata.obs`. IMO, it's fine to leave on the user-side: if they're willing to wait, that's alright. But if you prefer, a hard limit of 30-50 values can be used.

**Comment by euxhenh** [comment]
Sounds good! I will work on this. Adding it to the plot's header sounds right.

**Comment by VPetukhov** [comment]
Thank you so much!
